File: Javascript/26. Remove Duplicates from Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
	if (!nums) {
		return 0;
	}

	let k = 1;
	let slow = 0;

	for (let fast = 1; fast < nums.length; fast++) {
		if (nums[fast] !== nums[slow]) {
			k += 1;
			slow += 1;
			nums[slow] = nums[fast]
		}

	}
	return k;
};

File: Python/26. Remove Duplicates from Sorted Array.py
class Solution:
    def removeDuplicates(self, nums):
        if not nums:
            return 0

        k = 1  # Count of unique elements
        slow = 0  # Pointer to track the last unique element

        for fast in range(1, len(nums)):
            if nums[fast] != nums[slow]:
                slow += 1
                nums[slow] = nums[fast]
                k += 1

        return k
